pllug-presentation-system/presentation-data/domdocumentdivider.h

```diff
@@ -0,0 +1,32 @@
+#ifndef DOMDOCUMENTDIVIDER_H
+#define DOMDOCUMENTDIVIDER_H
+
+#include <QStringList>
+#include <QString>
+#include <QFile>
+#include <QVector>
+#include <QtWidgets/QMessageBox>
+#include <QDebug>
+#include <QDomDocument>
+
+#include "documentimportstrategy.h"
+#include "slide.h"
+#include "presentationelementfactory.h"
+#include "presentation.h"
+
+class DomDocumentDivider : public DocumentImportStrategy
+{
+public:
+    virtual std::unique_ptr<Presentation> import(const QByteArray &inputText) override;
+
+private:
+    bool isSeparator(const QDomElement &element);
+    bool isHeader(const QDomElement &element);
+
+    QVector<QDomDocument> divideBySeparators(const QDomDocument &htmlDocument);
```

Comment:
Do we need to do something about the return types in this class' methods to avoid unnecessary copies?